ios/Classes/components/applePay/BaseApplePayComponent.swift

```diff
@@ -7,6 +7,17 @@ class BaseApplePayComponent {
         preconditionFailure("This method must be overridden")
     }
     
+    func handlePaymentEvent(paymentEventDTO: PaymentEventDTO) {}
+    
+    func finalizeAndDismissComponent(success: Bool, completion: @escaping (() -> Void)) {
+        applePayComponent?.finalizeIfNeeded(with: success) { [weak self] in
+            completion()
```

Comment:
It feels calling completion before the top view controller is dismissed could lead to unecpected behavior if the completion closure involves any navigation changes. I'd suggest to think moving the call inside the second nested completion block (when the view controller gets dismissed)